platform-sdk/platform-apps/tests/ISSTestingTool/src/main/java/com/swirlds/demo/iss/ISSTestingToolState.java

```diff
@@ -95,52 +110,55 @@ public void initState(InitTrigger trigger, Platform platform) {
                 platformContext.getMerkleCryptography(),
                 () -> DEFAULT_PLATFORM_STATE_FACADE.roundOf(this));
 
+        final var schema = new V0660ISSTestingToolSchema();
+        schema.statesToCreate().stream()
+                .sorted(Comparator.comparing(StateDefinition::stateId))
+                .forEach(def -> {
+                    super.initializeState(new StateMetadata<>(ISS_SERVICE_NAME, schema, def));
+                });
+
         // since the test occurrences are relative to the genesis timestamp, the data only needs to be parsed at genesis
         if (trigger == InitTrigger.GENESIS) {
             final ISSTestingToolConfig testingToolConfig =
                     platform.getContext().getConfiguration().getConfigData(ISSTestingToolConfig.class);
 
             this.plannedIssList = testingToolConfig.getPlannedISSs();
             this.plannedLogErrorList = testingToolConfig.getPlannedLogErrors();
-            writeObjectByChildIndex(PLANNED_ISS_LIST_INDEX, plannedIssList);
-            writeObjectByChildIndex(PLANNED_LOG_ERROR_LIST_INDEX, plannedLogErrorList);
+
+            final WritableStates writableStates = getWritableStates(ISS_SERVICE_NAME);
+
+            final WritableQueueState<PlannedIss> plannedIssState = writableStates.getQueue(PLANNED_ISS_LIST_STATE_ID);
+            plannedIssList.forEach(plannedIssState::add);
+
+            final WritableQueueState<PlannedLogError> plannedLogErrorState =
+                    writableStates.getQueue(PLANNED_LOG_ERROR_LIST_STATE_ID);
+            plannedLogErrorList.forEach(plannedLogErrorState::add);
+
+            ((CommittableWritableStates) writableStates).commit();
         } else {
-            final StringLeaf runningSumLeaf = getChild(RUNNING_SUM_INDEX);
-            if (runningSumLeaf != null) {
-                runningSum = Long.parseLong(runningSumLeaf.getLabel());
-            }
-            final StringLeaf genesisTimestampLeaf = getChild(GENESIS_TIMESTAMP_INDEX);
-            if (genesisTimestampLeaf != null) {
-                genesisTimestamp = Instant.parse(genesisTimestampLeaf.getLabel());
+            final ReadableStates readableStates = getReadableStates(ISS_SERVICE_NAME);
+
+            final ReadableSingletonState<Long> runningSumState = readableStates.getSingleton(RUNNING_SUM_STATE_ID);
+            final Long runningSum = runningSumState.get();
+            if (runningSum != null) {
```

Comment:
Can it actually be null? The singleton state is always initialized. Does it ever accept a null value?